sklearn/preprocessing/label.py

```diff
@@ -37,6 +37,125 @@
 ]
 
 
+def _encode_numpy(values, uniques=None, encode=False):
+    if uniques is None:
+        if encode:
+            uniques, encoded = np.unique(values, return_inverse=True)
+            return uniques, encoded
+        else:
+            # unique sorts
+            return np.unique(values)
+    if encode:
+        diff = _encode_check_unknown(values, uniques)
+        if diff:
+            raise ValueError(
+                "y contains previously unseen labels: %s" % str(diff))
+        encoded = np.searchsorted(uniques, values)
+        return uniques, encoded
+    else:
+        return uniques
```

Comment:
Should "Unsorted categories are not supported for numerical categories" be validated and raised here?

Reply:
Otherwise please note in the `_encode` docstring that this is not ensured, but is an assumption.

Reply:
It could be validated here (and it might be easier to follow), but in principle this will give (a little bit) more overhead as it is not needed to do that on each transform (encoding) if it is ensured in fit. 
So for now making this more clear in the docstring.